src/parser/error/internal.rs

```diff
@@ -12,15 +12,81 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use nom::error::ErrorKind;
+use super::ErrorKind;
+use crate::parser::ErrorInput;
+use nom::error::{ErrorKind as NomErrorKind, ParseError};
+
+/// Internal-only error that should be converted to an [`Error`](super::Error)
+/// before being returned in a public API.
```

Comment:
Maybe add a little bit about _why_?

- why there are two error types
- why the conversion should happen when it does